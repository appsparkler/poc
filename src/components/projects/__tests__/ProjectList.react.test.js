import React from 'react'
import {Provider} from 'react-redux'
import {BrowserRouter} from 'react-router-dom'
import ProjectList, {useEffectForFetchingAndStoringProjects}
  from '../ProjectList.react'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'

it('should correctly call dispatch', () => {
  const dispatch = jest.fn()
  useEffectForFetchingAndStoringProjects(dispatch)
  expect(dispatch).toHaveBeenCalled()
})

it('should render correctly', () => {
  expect.assertions(1)
  const mockStore = configureStore()
  const project1 = {
    id: 2,
    title: 'Hello World',
    content: 'hello world...',
  }
  const store = mockStore({
    project: {
      projects: [project1, {...project1, id: 3}],
    },
  })
  const component = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <ProjectList />
        </Provider>
      </BrowserRouter>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
