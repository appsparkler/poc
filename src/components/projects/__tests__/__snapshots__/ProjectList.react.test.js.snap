// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="project-list section"
  style={
    Object {
      "maxHeight": "500px",
      "overflow": "auto",
      "padding": "0 20px",
    }
  }
>
  <a
    href="/project/2"
    onClick={[Function]}
  >
    <div>
      <div
        className="project-summary card"
      >
        <div
          className="card-content grey-text text-darken-3"
        >
          <span
            className="card-title"
          >
            Hello World
          </span>
          <p>
            Posted By Net Ninja
          </p>
          <p
            className="grey-text"
          >
            3rd September, 2am
          </p>
        </div>
      </div>
    </div>
  </a>
  <a
    href="/project/3"
    onClick={[Function]}
  >
    <div>
      <div
        className="project-summary card"
      >
        <div
          className="card-content grey-text text-darken-3"
        >
          <span
            className="card-title"
          >
            Hello World
          </span>
          <p>
            Posted By Net Ninja
          </p>
          <p
            className="grey-text"
          >
            3rd September, 2am
          </p>
        </div>
      </div>
    </div>
  </a>
</div>
`;
